architecture_discussion.md

```diff
@@ -0,0 +1,51 @@
+We have basically 3 design choices to implement a Jupyter Notebook Operator. The main goal is to implement an operator with some level 5 capabilities, i.e. the operator should execute one action to maintain running notebooks intelligently. 
+
+
+### 1. JupyterHub from ODH
+This [JupyterHub](https://github.com/jupyterhub/jupyterhub) implementation utilizes [kubespawner](https://github.com/jupyterhub/kubespawner) and it's hook system to start pods and do adjustments. 
+
+**PRO**
+* It's based on a lively community. 
+* RBAC for accessing the notebook spawner and server UI 
+* Easy to implement default profiles and 
+* Has some prometheus metrics built-in
+* Python based
+ 
+**CON** 
+* It does not follow operator best practices (no CRD, stateful with DB)
+* No way to leverage operator-sdk 
+* Users can't `oc rsh` into their pod.
```

Comment:
This is partially solved by a web UI terminal in Jupyter UI. If you think about this - some users who are only using JupyterHub might not even know/care it is running on OpenShift

Reply:
i agree with vpavlin, I don't think the inability to use `oc rsh` is a big deal for JH users since they have a terminal in the web ui available

Reply:
I think the main usecase for rsh was for pvc-related tasks. So if we incorporate the pvc recovery playbook in somehow this requirement can probably be dropped.